### ISSUE TITLE
Should we add a transpose for the connection matrix?

### ISSUE THREAD

**Issue by hutuxian** [CRIPAC-DIG/SR-GNN#12]
```
u_sum_in = np.sum(u_A, 0)
u_sum_in[np.where(u_sum_in == 0)] = 1
u_A_in = np.divide(u_A, u_sum_in)
u_sum_out = np.sum(u_A, 1)
u_sum_out[np.where(u_sum_out == 0)] = 1
u_A_out = np.divide(u_A.transpose(), u_sum_out)
```
It seems we should add a transpose for u_A_in and u_A_out for normalization. Does it matter?

**Comment by tangrizzly** [comment]
The result after normalization is a little bit better.

**Comment by tangrizzly** [comment]
u_A_in和u

**Comment by tangrizzly** [comment]
Sorry to misunderstand your question.
<img width="437" alt="connection matrix" src="https://user-images.githubusercontent.com/25576583/55279931-07b67380-531f-11e9-9bf9-9e2285b641d1.png">
As shown in the image, u_A_out is the transposed matrix of u_A_in. Since a horizontal vector in the connection matrix represents the connection relationship of one node.

**Comment by hutuxian** [comment]
Yeah, but what confuses me is it seems the code and the figure in the paper doesn't match.
Take the above figure as an example, some of the corresponding variables in the code looks like:
<img width="348" alt="connection matrix" src="https://user-images.githubusercontent.com/11195205/55285804-3f77f680-53c5-11e9-991f-b77c31be7742.png">
So i say maybe we should add a transpose in u_A_in and u_A_out?

**Comment by tangrizzly** [comment]
We have added a transpose in u_A after concatenating the u_A_in and u_A_out. You can refer to our code in `utils.py`.
<img width="513" alt="Screen Shot 2019-03-31 at 9 10 32 AM" src="https://user-images.githubusercontent.com/25576583/55285931-2bb49c00-5395-11e9-909b-d05b01edb0f9.png">

**Comment by hutuxian** [comment]
Oh, I guess what you refer to is the pytorch code, and what I showed before  is the tensorflow code, but I didn't see the transpose. I hope this time we are on the same page.

**Comment by tangrizzly** [comment]
Thank you for pointing out. I think it does need a transpose. 

Our original code is the PyTorch version and the Tensorflow version is reproduced by another student after the paper was accepted. I will recheck the Tensorflow code to make sure it is consistent with the Pytorch version as well as the paper. And if you are in a hurry, you can refer to our PyTorch code first.

**Comment by hutuxian** [comment]
Got it, thanks for your patient reply.
